### PR TITLE
Fix error message

### PR DESCRIPTION
**Pull request by bells17**
#### What type of PR is this?

/kind cleanup
/sig storage

#### What this PR does / why we need it:

Fixed a error message.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go [hunk 1]
```diff
@@ -679,7 +679,7 @@ func TestVolumeBinding(t *testing.T) {
 			for i, nodeInfo := range nodeInfos {
 				gotStatus := p.Filter(ctx, state, item.pod, nodeInfo)
 				if !reflect.DeepEqual(gotStatus, item.wantFilterStatus[i]) {
-					t.Errorf("filter status does not match for node %q, got: %v, want: %v", nodeInfo.Node().Name, gotStatus, item.wantFilterStatus)
+					t.Errorf("filter status does not match for node %q, got: %v, want: %v", nodeInfo.Node().Name, gotStatus, item.wantFilterStatus[i])
```
Comment (kerthcet) [inline pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go:682]: Better with `cmp.Diff` here, like:
```
if diff := cmp.Diff(item.wantFilterStatus[i], gotStatus); diff != "" {
    t.Errorf("filter status does not match for node %q, (-want,+got):\n%s", nodeInfo.Node().Name, diff)
}
```

**Comment by kerthcet** [comment]
thanks @bells17 
/lgtm

**Comment by kerthcet** [comment]
oh, please squash the commits.

**Comment by bells17** [comment]
Sorry. I squashed commits.

**Review by sanposhiho — COMMENTED** [review]
Could you add the same improvement on L657-L659?

**Comment by bells17** [comment]
@sanposhiho Thank you for reviewing. I modified it to use cmd.Diff on L657-L659.
example output is as follows:

```
make test WHAT=./pkg/scheduler/framework/plugins/volumebinding | grep -v "Mutation detector is enabled, this will result in memory leakage."
+++ [0513 16:44:16] Building go targets for linux/amd64
    k8s.io/kubernetes/hack/make-rules/helpers/go2make (non-static)
+++ [0513 16:45:04] Running tests without code coverage and with -race
--- FAIL: TestVolumeBinding (0.21s)
    --- FAIL: TestVolumeBinding/pod_has_not_pvcs (0.11s)
        volume_binding_test.go:623: Feed testing data and wait for them to be synced
        volume_binding_test.go:637: Start informer factory after initialization
        volume_binding_test.go:640: Wait for all started informers' cache were synced
        volume_binding_test.go:643: Verify
        volume_binding_test.go:654: Verify: call PreFilter and check status
        volume_binding_test.go:657: filter prefilter status does not match: (-want,+got):
              (*framework.Status)(
            -   &{
            -           code:    s"UnschedulableAndUnresolvable",
            -           reasons: []string{`persistentvolumeclaim "pvc-a" not found`},
            -   },
            +   nil,
              )
        volume_binding_test.go:664: Verify: check state after prefilter phase
        volume_binding_test.go:677: Verify: call Filter and check status
        volume_binding_test.go:685: Verify: Score
    --- FAIL: TestVolumeBinding/PVC_does_not_exist (0.11s)
        volume_binding_test.go:623: Feed testing data and wait for them to be synced
        volume_binding_test.go:637: Start informer factory after initialization
        volume_binding_test.go:640: Wait for all started informers' cache were synced
        volume_binding_test.go:643: Verify
        volume_binding_test.go:654: Verify: call PreFilter and check status
        volume_binding_test.go:657: filter prefilter status does not match: (-want,+got):
              (*framework.Status)(
            -   &{
            -           code:    s"UnschedulableAndUnresolvable",
            -           reasons: []string{`persistentvolumeclaim "pvc-a" not found - test`},
            -   },
            +   &{
            +           code:    s"UnschedulableAndUnresolvable",
            +           reasons: []string{`persistentvolumeclaim "pvc-a" not found`},
            +   },
              )
FAIL
FAIL    k8s.io/kubernetes/pkg/scheduler/framework/plugins/volumebinding 19.552s
FAIL
```

File: pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go [hunk 2]
```diff
@@ -654,8 +653,8 @@ func TestVolumeBinding(t *testing.T) {
 
 			t.Logf("Verify: call PreFilter and check status")
 			_, gotPreFilterStatus := p.PreFilter(ctx, state, item.pod)
-			if !reflect.DeepEqual(gotPreFilterStatus, item.wantPreFilterStatus) {
-				t.Errorf("filter prefilter status does not match: %v, want: %v", gotPreFilterStatus, item.wantPreFilterStatus)
+			if diff := cmp.Diff(item.wantPreFilterStatus, gotPreFilterStatus); diff != "" {
+				t.Errorf("filter prefilter status does not match: (-want,+got):\n%s", diff)
```
Comment (sanposhiho) [inline pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go:657]: It seems to be a typo.

```suggestion
				t.Errorf("prefilter status does not match: (-want,+got):\n%s", diff)
```

**Review by sanposhiho — APPROVED** [review]
Thanks.

/lgtm

**Comment by bells17** [comment]
@sanposhiho I'm took in your suggestion. Would you review again?

**Comment by sanposhiho** [comment]
(Please squash)

**Comment by bells17** [comment]
@sanposhiho Sorry, I squashed commits.

**Comment by sanposhiho** [comment]
For approval
/assign @gnufied

**Comment by pohly** [comment]
My 2 cents for future test enhancements: to get a nice diff output for complex structures, use `stretchr/testify/[require|assert].Equal`. It's okay for Kubernetes unit tests and can be used with less code than `cmp.Diff`.

**Comment by bells17** [comment]
@pohly Thank you for reviewing. I modified to use "github.com/stretchr/testify/assert".
Would you review this again?

example output is as follows:
```
$ make test WHAT=./pkg/scheduler/framework/plugins/volumebinding | grep -v "Mutation detector is enabled, this will result in memory leakage."
+++ [0805 00:49:40] Building go targets for linux/amd64
    k8s.io/kubernetes/hack/make-rules/helpers/go2make (non-static)
+++ [0805 00:50:01] Running tests without code coverage and with -race
--- FAIL: TestVolumeBinding (0.32s)
    --- FAIL: TestVolumeBinding/PVC_does_not_exist (0.11s)
        volume_binding_test.go:623: Feed testing data and wait for them to be synced
        volume_binding_test.go:637: Start informer factory after initialization
        volume_binding_test.go:640: Wait for all started informers' cache were synced
        volume_binding_test.go:643: Verify
        volume_binding_test.go:654: Verify: call PreFilter and check status
        volume_binding_test.go:656: 
                Error Trace:    volume_binding_test.go:656
                Error:          Not equal: 
                                expected: &framework.Status{code:3, reasons:[]string{"persistentvolumeclaim \"pvc-a\" not found test"}, err:error(nil), failedPlugin:""}
                                actual  : &framework.Status{code:3, reasons:[]string{"persistentvolumeclaim \"pvc-a\" not found"}, err:error(nil), failedPlugin:""}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -3,3 +3,3 @@
                                  reasons: ([]string) (len=1) {
                                -  (string) (len=44) "persistentvolumeclaim \"pvc-a\" not found test"
                                +  (string) (len=39) "persistentvolumeclaim \"pvc-a\" not found"
                                  },
                Test:           TestVolumeBinding/PVC_does_not_exist
FAIL
FAIL    k8s.io/kubernetes/pkg/scheduler/framework/plugins/volumebinding 21.674s
FAIL
make: *** [test] Error 1
Makefile:186: recipe for target 'test' failed
```

**Comment by pohly** [comment]
I like the version with stretchr better - no surprise, as I suggested it :grin: 

But therefore I'd like others to re-add their LGTM. @sanposhiho, @kerthcet?

**Comment by pohly** [comment]
Let me point out one advantage of assert.Equal: if the objects are large, then some common fields will not be shown by cmp.Diff, which can make the part that gets printed harder to understand. assert.Equal avoids this by showing everything (full objects and diff).

**Pull request merged**